cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -105,12 +105,15 @@ struct ARROW_EXPORT VarianceOptions : public FunctionOptions {
 /// By default, returns the median value.
 struct ARROW_EXPORT QuantileOptions : public FunctionOptions {
   /// Interpolation method to use when quantile lies between two data points
+  /// TDIGEST is useful to approximate quantiles from large volume inputs.
+  /// It has constant memory footprint, but lower accuracy.
   enum Interpolation {
     LINEAR = 0,
     LOWER,
     HIGHER,
     NEAREST,
     MIDPOINT,
+    TDIGEST,
```

Comment:
I'm curious whether this is best exposed as an interpolation kind for the "quantile" function, or a separate function altogether. Are there precedents in other libraries or database engines?

It seems R uses a separate function:
https://www.rdocumentation.org/packages/tdigest/versions/0.3.0/topics/tdigest

cc @nealrichardson @michalursa  for opinion.

Reply:
Matlab quantile has a `method` parameter, which can be `exact` (linear interpolation) or `approximate` (t-digest).
https://www.mathworks.com/help/stats/quantile.html#mw_06fb6551-ec1a-48f6-a690-f790dfaa9849

Reply:
FWIW that tdigest R function is not part of base R, it is a contributed package.

R's quantile function also supports multiple methods, 9 in fact, via the `type` parameter: https://stat.ethz.ch/R-manual/R-devel/library/stats/html/quantile.html

None of these are approximate in the same way as this, but it's a further argument that it could be a function parameter rather than a separate function. TBH I don't know how much it matters, as a compute API consumer I can make either work. It's marginally easier if they're separate functions rather than having to mess with `FunctionOptions` to select them--are maybe there are downsides to that way?

Reply:
Note that the t-digest method also has its own parameters (compression and delta). If we ever want to expose them, it would probably be less confusing as a separate function.

Reply:
Sounds like separate function makes the most sense unless there's a cost to having an additional function.

Reply:
> Note that the t-digest method also has its own parameters (compression and delta). If we ever want to expose them, it would probably be less confusing as a separate function.

Good point. I will separate it into another function.

Reply:
Add separated `TDigest` kernel.

Reply:
@pitrou I've separated TDigest kernel, please review, thanks.